tests/tests/htlc_tests.cpp

```diff
@@ -0,0 +1,408 @@
+
+std::vector<unsigned char> hash_it(std::vector<unsigned char> preimage)
+{
+   // convert the preimage to a char array
+   unsigned char char_array[preimage.size()];
+   for(unsigned int i = 0; i < preimage.size(); ++i)
+      char_array[i] = preimage[i];
+	fc::sha256 hash = fc::sha256::hash((char*)char_array, preimage.size());
+	std::vector<unsigned char> ret_val(hash.data_size());
+	char* data = hash.data();
+	for(size_t i = 0; i < hash.data_size(); i++)
+	{
+		ret_val[i] = data[i];
+	}
+	return ret_val;
+}
+
+void set_committee_parameters(database_fixture* db_fixture)
+{
+   // set the committee parameters
+   db_fixture->db.modify(db_fixture->db.get_global_properties(), [](global_property_object& p) {
+      // htlc options
+      graphene::chain::htlc_options params;
+      params.max_preimage_size = 1024;
+      params.max_timeout_secs = 60 * 60 * 24 * 28;
+      p.parameters.extensions.value.updatable_htlc_options = params;
+      // htlc operation fees
+      p.parameters.current_fees->get<htlc_create_operation>().fee = 2 * GRAPHENE_BLOCKCHAIN_PRECISION;
+      p.parameters.current_fees->get<htlc_create_operation>().fee_per_day = 2 * GRAPHENE_BLOCKCHAIN_PRECISION;
+   });
+}
+
+void advance_past_hardfork(database_fixture* db_fixture)
+{
+   db_fixture->generate_blocks(HARDFORK_HTLC_TIME);
+   set_committee_parameters(db_fixture);
+   set_expiration(db_fixture->db, db_fixture->trx);
+}
+
+BOOST_AUTO_TEST_CASE( htlc_before_hardfork )
+{
+   ACTORS((alice)(bob));
+
+   int64_t init_balance(100000);
+
+   transfer( committee_account, alice_id, graphene::chain::asset(init_balance) );
+
+   uint16_t key_size = 256;
+   std::vector<unsigned char> pre_image(256);
+   generate_random_preimage(key_size, pre_image);
+   std::vector<unsigned char> key_hash = hash_it(pre_image);
+
+   graphene::chain::htlc_id_type alice_htlc_id;
+   // cler everything out
+   generate_block();
+   trx.clear();
+   // Alice tries to put a contract on the blockchain
+   {
+      graphene::chain::htlc_create_operation create_operation;
+
+      create_operation.amount = graphene::chain::asset( 10000 );
+      create_operation.destination = bob_id;
+      create_operation.seconds_in_force = 60;
+      create_operation.key_hash = key_hash;
+      create_operation.hash_type = graphene::chain::hash_algorithm::sha256;
+      create_operation.key_size = key_size;
+      create_operation.source = alice_id;
+      trx.operations.push_back(create_operation);
+      sign(trx, alice_private_key);
+      GRAPHENE_CHECK_THROW(PUSH_TX(db, trx, ~0), fc::exception);
+   }
+   // attempt to create a proposal that contains htlc stuff
+   {
+
+      graphene::chain::committee_member_update_global_parameters_operation param_op;
+
+      graphene::chain::proposal_create_operation create_operation;
+
+      create_operation.fee_paying_account = committee_account;
+      create_operation.review_period_seconds = 60 * 60 * 48;
+      create_operation.proposed_ops.emplace_back(param_op);
+   }
```

Comment:
Add check for other htlc operations inside proposal. And for update_parameters with fees for new operations.

Reply:
Merged in your changes that include htlc proposal tests